testing/events_test.go

```diff
@@ -0,0 +1,223 @@
+							Value: "42",
+						},
+						{
+							Key:   channeltypes.AttributeKeySrcPort,
+							Value: "srcPort",
+						},
+						{
+							Key:   channeltypes.AttributeKeySrcChannel,
+							Value: "srcChannel",
+						},
+						{
+							Key:   channeltypes.AttributeKeyDstPort,
+							Value: "dstPort",
+						},
+						{
+							Key:   channeltypes.AttributeKeyDstChannel,
+							Value: "dstChannel",
+						},
+						{
+							Key:   channeltypes.AttributeKeyTimeoutHeight,
+							Value: "1-2",
+						},
+						{
+							Key:   channeltypes.AttributeKeyTimeoutTimestamp,
+							Value: "1000",
+						},
+					},
+				},
+				{
+					Type: "yyy",
+				},
+				{
+					Type: channeltypes.EventTypeSendPacket,
+					Attributes: []abci.EventAttribute{
+						{
+							Key:   channeltypes.AttributeKeyData,
+							Value: "data2",
+						},
+						{
+							Key:   channeltypes.AttributeKeySequence,
+							Value: "43",
+						},
+						{
+							Key:   channeltypes.AttributeKeySrcPort,
+							Value: "srcPort",
+						},
+						{
+							Key:   channeltypes.AttributeKeySrcChannel,
+							Value: "srcChannel",
+						},
+						{
+							Key:   channeltypes.AttributeKeyDstPort,
+							Value: "dstPort",
+						},
+						{
+							Key:   channeltypes.AttributeKeyDstChannel,
+							Value: "dstChannel",
+						},
+						{
+							Key:   channeltypes.AttributeKeyTimeoutHeight,
+							Value: "1-3",
+						},
+						{
+							Key:   channeltypes.AttributeKeyTimeoutTimestamp,
+							Value: "1001",
+						},
+					},
+				},
+			},
+			expectedPackets: []channeltypes.Packet{
+				{
+					Sequence:           42,
+					SourcePort:         "srcPort",
+					SourceChannel:      "srcChannel",
+					DestinationPort:    "dstPort",
+					DestinationChannel: "dstChannel",
+					Data:               []byte("data1"),
+					TimeoutHeight: types.Height{
+						RevisionNumber: 1,
+						RevisionHeight: 2,
+					},
+					TimeoutTimestamp: 1000,
+				},
+				{
+					Sequence:           43,
+					SourcePort:         "srcPort",
+					SourceChannel:      "srcChannel",
+					DestinationPort:    "dstPort",
+					DestinationChannel: "dstChannel",
+					Data:               []byte("data2"),
+					TimeoutHeight: types.Height{
+						RevisionNumber: 1,
+						RevisionHeight: 3,
+					},
+					TimeoutTimestamp: 1001,
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+			assert := assert.New(t)
+
+			allPackets, err := ibctesting.ParsePacketsFromEvents(tt.events)
+
+			if tt.expectedError != "" {
+				assert.ErrorContains(err, tt.expectedError)
+			} else {
+				require.NoError(err)
+				assert.Equal(tt.expectedPackets, allPackets)
+			}
+
+			firstPacket, err := ibctesting.ParsePacketFromEvents(tt.events)
+
+			if tt.expectedError != "" {
+				assert.ErrorContains(err, tt.expectedError)
+			} else {
+				require.NoError(err)
+				assert.Equal(tt.expectedPackets[0], firstPacket)
+			}
+		})
+	}
```

Comment:
I think this could written like this (assuming that `tests` is renamed to `testCases`):


```golang
for _, tc := range testCases {
  t.Run(tc.name, func(t *testing.T) {
    allPackets, err := ibctesting.ParsePacketsFromEvents(tc.events)

    if tc.expectedError == "" {
      require.NoError(t, err)
      require.Equal(t, tc.expectedPackets, allPackets)
    } else {
      require.ErrorContains(t, err, tc.expectedError)
    }

    firstPacket, err := ibctesting.ParsePacketFromEvents(tc.events)

    if tc.expectedError == "" {
      require.NoError(t, err)
      require.Equal(t, tc.expectedPackets[0], firstPacket)
    } else {
      require.ErrorContains(t, err, tc.expectedError)
    }
   })
}
```

So that:
- We don't need to import `assert`.
- We execute first the checks for the success case.